### PR TITLE
Only consider known layers in FeatureIndex#query

### LINKED ISSUE #4417: map.queryRenderedFeatures went wrong  when  circle-radius in data-driven style
**mapbox-gl-js version 0.33.1**:

### Steps to Trigger Behavior

 1. open https://jsbin.com/jagojuz/edit?html,console,output
 2. open the browser's `console`
 3. click on the red point

### Expected Behavior
image

### Actual Behavior
image

When I change:
```js
 map.addLayer({
      id: 'test',
      type: 'circle',
      source: 'testgeo',
      paint: {
        'circle-radius': {
          property: 'size',
          type: 'identity'
        },
        'circle-color': '#f00'
      }
    });
```
to
```js
 map.addLayer({
      id: 'test',
      type: 'circle',
      source: 'testgeo',
      paint: {
        'circle-radius':10,
        'circle-color': '#f00'
      }
    });
```
everything will be ok.

### PR DESCRIPTION
**Pull request by anandthakker**
Closes #4417

### CODE REVIEW

File: src/data/feature_index.js [hunk 1]
```diff
@@ -41,6 +41,14 @@ class FeatureIndex {
             this.rawTileData = rawTileData;
             this.bucketLayerIDs = serialized.bucketLayerIDs;
             this.paintPropertyStatistics = serialized.paintPropertyStatistics;
+
+            // make a list of layer ids known to this FeatureIndex
+            this.knownLayers = {};
+            for (const index in this.bucketLayerIDs) {
+                for (const id of this.bucketLayerIDs[index]) {
+                    this.knownLayers[id] = true;
```
Comment (lucaswoj) [inline src/data/feature_index.js]: I worry that the word "known" might make less sense once the context of this PR has faded. How 'bout just `layerIDs`? 

 - `bucketLayerIDs` -> layer ids grouped by bucket
 - `layerIDs` -> plain old layer ids.

File: src/data/feature_index.js [hunk 2]
```diff
@@ -111,6 +119,8 @@ class FeatureIndex {
         // expanded by to include these features.
         let additionalRadius = 0;
         for (const id in styleLayers) {
+            if (!this.knownLayers[id]) continue;
```
Comment (lucaswoj) [inline src/data/feature_index.js]: Is this code especially hot? Could we get by with a less stateful & slightly less efficient `this.hasLayer(id)` method?

File: test/integration/query-tests/regressions/mapbox-gl-js#4417/style.json [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        }
+      }
+    },
+    {
+      "id": "dummy",
+      "type": "circle",
+      "source": "dummy"
+    }
+  ]
+}
```
Comment (lucaswoj) [inline test/integration/query-tests/regressions/mapbox-gl-js#4417/style.json]: ❤️ tests 😂

**Comment by jfirebaugh** [comment]
Can you add some context to the commit summary -- what the trigger for the issue was and how the change resolves it?

**Comment by anandthakker** [comment]
@jfirebaugh @lucaswoj updated per notes

**Pull request merged**